File: ParallelBatchEval/Py/QAP.py
import numpy as np
import pickle
import os.path

# read from input file
def read_input(instname):
    filename="../../instances/nug/"+instname+".dat"
    with open(filename, 'r') as f:
        content = f.read()
        lines = content.split('\n')
        dim = np.int64(lines[0])
        A = [[0 for x in range(dim)] for x in range(2 * dim)]
        i = 0

        for l in lines[1:]:
            if not l.strip():
                continue
            if l.startswith('EOF'):
                break
            else:
                A[i] = [int(x) for x in l.split()]
                i += 1

    flow = np.array(A[0:dim])
    dist = np.array(A[dim:2 * dim])

    return dist,flow,dim

File: ParallelBatchEval/Julia/QAP.jl
module QAP
    using DelimitedFiles

	dim = Int(0)
	moves = Array{Int,2}
    flow = Array{Int,2}
    dist = Array{Int,2}

	export read_in_QAP,evaluate_QAP
	export dim,flow,dist

#####################################
	function read_in_QAP(instname)
		filename = string("../../instances/qaplib/",instname,".dat")

        # read from file...
        file = open(filename)
        global dim = parse(Int,readline(file, keep=false))
        mat = readdlm(file,Int)
    	global flow=mat[1:dim,:]
    	global dist=mat[dim+1:2*dim,:]
    end
#####################################

#####################################
	function evaluate_QAP(perm,dist,flow,dim)
        cost = 0
		for i=1:dim,j=1:dim
			cost += flow[i,j]*dist[perm[i],perm[j]]
        end
		return cost
	end
#####################################

end
